### PR TITLE
Connect api

### PR DESCRIPTION
**Pull request by paulinagonzalezc**
# Bookstore: connect to API

In this project, I connected my React-Redux application to an existing API to create and remove books in a remote server

General requirements

* Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config)
* Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)
* Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)

Project requirements

* Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API
* Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls
* Fetch book list from Bookstore API on Books page load
* Refactor your add book and remove book features to persist your changes in the server
* Add book and remove book should work in the same way after the refactor
* No styling is required

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,38 +1,31 @@
...
-        action.book,
-      ];
-    case REMOVE:
-      return state.filter((book) => book.id !== action.book.id);
+    case `${ADD}/fulfilled`:
+      return state.concat(action.meta.arg);
+    case `${REMOVE}/fulfilled`:
+      return state.filter((book) => book.item_id !== action.meta.arg);
+    case `${GET}/fulfilled`:
+    // console.log('chewie', action.payload);
```
Comment (danmainah) [inline src/redux/books/books.js]: - [x] @paulinagonzalezc great work on your reducer function but you forgot to remove the commented line, it's not good practice to have it in our codebase.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -47,3 +40,31 @@ export const removeBook = (book) => ({
...
+
+export const createBook = createAsyncThunk(ADD, async (book) => {
+  const response = await fetch(url, {
+    method: 'POST',
+    body: JSON.stringify(book),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  console.log(response);
```
Comment (danmainah) [inline src/redux/books/books.js]: - [x] @paulinagonzalezc great work on your rcreateBook function but you forgot to remove the console.log, its not good practice to have it in our codebase

File: src/redux/books/books.js [hunk 3]
```diff
@@ -47,3 +40,31 @@ export const removeBook = (book) => ({
...
+});
+
+export const deleteBook = createAsyncThunk(REMOVE, async (itemId) => {
+  const response = await fetch(`${url}/${itemId}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  console.log(response);
```
Comment (danmainah) [inline src/redux/books/books.js]: - [x] @paulinagonzalezc great work on your deleteBook function but you forgot to remove the console.log, its not good practice to have it in our codebase

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @paulinagonzalezc , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 1 above)
Comment (paulinagonzalezc) [reply src/redux/books/books.js]: Thank you @danmainah, I will remove it.

File: src/redux/books/books.js (hunk 2 above)
Comment (paulinagonzalezc) [reply src/redux/books/books.js]: Thank you @danmainah, I will remove it.

File: src/redux/books/books.js (hunk 3 above)
Comment (paulinagonzalezc) [reply src/redux/books/books.js]: Thank you @danmainah, I will remove it.

**Review by Ismailco — APPROVED** [review]
### Hi @paulinagonzalezc, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**